routes/api_v2_0.py

```diff
@@ -441,6 +447,230 @@ def class4_data(
+                var_units = var_data.get("units", "")
+
+                if var_key not in all_quiver_vars:
+                    all_quiver_vars[var_key] = {
+                        "id": var_key,
+                        "name": var_name,
+                        "units": var_units,
+                        "datasets": [dataset_key],
+                    }
+                else:
+                    if dataset_key not in all_quiver_vars[var_key]["datasets"]:
+                        all_quiver_vars[var_key]["datasets"].append(dataset_key)
+
+    return {"data": list(all_quiver_vars.values())}
+
+
+@router.get("/datasets/filter/quiver_variable")
+def filter_datasets_by_quiver_variable(
+    quiver_variable: str = Query(description="Quiver variable to filter by"),
+    dataset_ids: str = Query(description="Comma-separated dataset IDs to filter from"),
+):
+    """
+    Returns matching dataset IDs and quiver variable info.
+    """
+    dataset_id_list = [id.strip() for id in dataset_ids.split(",") if id.strip()]
+
+    matching_datasets = []
+
+    for dataset_id in dataset_id_list:
+        config = DatasetConfig(dataset_id)
+        for var_key in config.vector_variables.keys():
+            var_data = config.vector_variables[var_key]
+            var_name = var_data.get("name", var_key)
+            if (
+                quiver_variable == var_name
+                and getattr(config, "model_class", "") == "Mercator"
+            ):
+                matching_datasets.append(
+                    {
+                        "id": dataset_id,
+                        "quiverVariable": quiver_variable,
+                    }
+                )
+
+    return {"datasets": matching_datasets}
+
+
+@router.get("/datasets/filter/depth")
+def filter_datasets_by_depth(
+    has_depth: bool = Query(description="Depth requirement: 'yes', 'no'"),
+    dataset_ids: str = Query(description="Comma-separated dataset IDs to filter from"),
+):
+    """
+    Returns only matching dataset IDs for depth filter.
+    """
+    dataset_id_list = [id.strip() for id in dataset_ids.split(",") if id.strip()]
+
+    matching_dataset_ids = []
+
+    for dataset_id in dataset_id_list:
+        config = DatasetConfig(dataset_id)
+        with SQLiteDatabase(config.url) as db:
+            dims = db.get_all_dimensions()
+            has_depth_dims = True if "depth" in dims else False
+            if has_depth == has_depth_dims:
+                matching_dataset_ids.append(dataset_id)
+
+    return {"dataset_ids": matching_dataset_ids}
+
+
+@router.get("/datasets/filter/date")
+def filter_datasets_by_date(
+    target_date: str = Query(description="Target date in ISO format"),
+    dataset_ids: str = Query(description="Comma-separated dataset IDs to filter from"),
+):
+    """
+    Returns only matching dataset IDs for date filter.
+    """
+    dataset_id_list = [id.strip() for id in dataset_ids.split(",") if id.strip()]
```

Comment:
If you specify the `dataset_ids` param as a list then you shouldn't have to decode the string assuming it's encoded correctly in the query.